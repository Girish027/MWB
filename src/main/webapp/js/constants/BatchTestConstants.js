
const PREDICTED_RUI = 'Predicted RUI';
const batchTestConstants = {
  DEFAULT_PAGE_SIZE: 100,

  // Filter Related
  TRANSCRIPTIONS_FILTER_NAME: 'Transcriptions',
  TRANSCRIPTIONS_FILTER_LIST: ['All', 'Misclassified'],
  INTENT_FILTER_NAME: 'Intent',
  FILTER_SELECTED_ALL: 'All',
  ROLLUP_INTENT_ROW_ID: 'Rollup Intent (RUI)',
  PREDICTED_RUI_1: `${PREDICTED_RUI} 1`,
  PREDICTED_RUI_2: `${PREDICTED_RUI} 2`,
  PREDICTED_RUI_3: `${PREDICTED_RUI} 3`,
  PREDICTED_RUI_4: `${PREDICTED_RUI} 4`,
  PREDICTED_RUI_5: `${PREDICTED_RUI} 5`,
};

export default batchTestConstants;
