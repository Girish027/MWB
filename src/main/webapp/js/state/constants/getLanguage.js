import config from 'config';
import * as english from './english';

export const getLanguage = () => {
  switch (config.lang) {
  case 'english':
  default:
    return english;
  }
};
