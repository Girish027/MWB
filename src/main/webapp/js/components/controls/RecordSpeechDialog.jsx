import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Dialog,
  StartRecordingIcon,
  StopRecordingIcon,
  PlayRecordingIcon,
  PrimaryDivider,
} from '@tfs/ui-components';
import * as appActions from 'state/actions/actions_app';
import Constants from 'constants/Constants';
import AudioPlayer from 'components/audio/AudioPlayer';
import AudioRecorder from 'components/audio/AudioRecorder';
import { getLanguage } from 'state/constants/getLanguage';
import Downloader from 'components/controls/Downloader';

const { DISPLAY_MESSAGES } = getLanguage();

export default class RecordSpeechDialog extends Component {
  constructor(props) {
    super(props);
    this.props = props;

    this.audioStates = {
      beforeRecording: 'beforeRecording',
      recording: 'recording',
      beforePlaying: 'beforePlaying',
      playing: 'playing',
    };

    this.subtileMap = {
      beforeRecording: 'START',
      recording: 'STOP',
      beforePlaying: 'PLAY',
      playing: 'STOP',
    };

    this.initialState = {
      isRunTestDisable: true,
      isPlayButtonEnabled: false,
      seconds: Constants.RECORDING_MAX_TIME_LIMIT_SEC,
      minutes: Constants.RECORDING_MAX_TIME_LIMIT_MIN,
      audioFile: '',
      audioSrc: '',
      audioState: this.audioStates.beforeRecording,
    };

    this.state = {
      ...this.initialState,
    };

    this.registerCallbacks = this.registerCallbacks.bind(this);
    this.onClickClose = this.onClickClose.bind(this);
    this.onClickStartRecord = this.onClickStartRecord.bind(this);
    this.onClickStopRecord = this.onClickStopRecord.bind(this);
    this.onClickRecordNew = this.onClickRecordNew.bind(this);
    this.onClickRunTest = this.onClickRunTest.bind(this);
    this.setAudioState = this.setAudioState.bind(this);
    this.renderChildren = this.renderChildren.bind(this);
    this.tick = this.tick.bind(this);

    this.timer = null;
    this.callbacks = {};
    this.styleOverride = {
      container: {
        width: '500px',
        height: '340px',
        display: 'grid',
        gridTemplateRows: '60px auto 60px',
      },
      childContainer: {
        marginTop: '10px',
        marginBottom: '10px',
      },
      cancel: {
        backgroundColor: 'unset',
        paddingLeft: '10px',
        paddingRight: '10px',
      },
      ok: {
        marginLeft: '10px',
        paddingLeft: '25px',
        paddingRight: '25px',
      },
    };
  }

  registerCallbacks(callbackItem) {
    this.callbacks = {
      ...this.callbacks,
      ...callbackItem,
    };
  }

  tick() {
    let { seconds } = this.state;
    if (seconds === 0) {
      clearInterval(this.timer);
      const {
        stopMicrophone = Constants.noop,
      } = this.callbacks;
      stopMicrophone();
    } else {
      this.setState({
        seconds: --seconds,
      });
    }
  }

  onClickClose() {
    const { dispatch } = this.props;
    // reset state
    this.setState(this.initialState);
    dispatch(appActions.modalDialogChange(null));
  }

  onClickStartRecord() {
    this.timer = setInterval(this.tick, 1000);
    this.setState({
      audioState: this.audioStates.recording,
    });
  }

  onClickStopRecord(audioFile) {
    clearInterval(this.timer);
    this.setState({
      isPlayButtonEnabled: true,
      isRunTestDisable: false,
      seconds: Constants.RECORDING_MAX_TIME_LIMIT_SEC,
      audioState: this.audioStates.beforePlaying,
      audioFile,
      audioSrc: URL.createObjectURL(audioFile),
    });
  }

  onClickRecordNew() {
    // reset state
    this.setState(this.initialState);
  }

  onClickRunTest() {
    const { dispatch, onClickRunTest } = this.props;
    const { audioFile } = this.state;
    if (audioFile) {
      //       dispatch(appActions.modalDialogChange({
      //         dispatch,
      //         type: Constants.DIALOGS.PROGRESS_DIALOG,
      //         message: Constants.TEST_IN_PROGRESS,
      //       }));
      dispatch(appActions.modalDialogChange(null));
      onClickRunTest(audioFile);
    } else {
      dispatch(appActions.displayWarningRequestMessage(DISPLAY_MESSAGES.audioFileNotAvailable));
    }
  }

  setAudioState(audioState) {
    this.setState({
      audioState,
    });
  }

  getFileName() {
    return `recording_${Date.now()}.wav`;
  }

  getActionSubtitle(addClass = 'blueText') {
    const { audioState } = this.state;
    let download = '';
    switch (audioState) {
    case this.audioStates.recording:
      addClass = 'blueText';
      break;
    case this.audioStates.beforePlaying:
    case this.audioStates.playing: {
      const { dispatch } = this.props;
      const { audioSrc } = this.state;
      download = (
        <span
          style={{
            paddingLeft: '10px',
          }}
        >
          <PrimaryDivider fill="#004C97" />
          <Downloader
            file={audioSrc}
            fileName={this.getFileName()}
            fileType={Constants.FILE_TYPE.RECORDING}
            dispatch={dispatch}
          >
            DOWNLOAD
          </Downloader>
        </span>
      );
    }
      break;
    default:
      break;
    }
    return (
      <span className={`speech-recording-span ${addClass}`}>
        {this.subtileMap[audioState]}
        {download}
      </span>
    );
  }

  renderChildren() {
    const { dispatch } = this.props;
    const {
      minutes,
      seconds,
      audioState,
      audioSrc,
    } = this.state;

    switch (audioState) {
    case this.audioStates.beforeRecording:
    case this.audioStates.recording:
      return (
        <span>
          {audioState === this.audioStates.recording
            ? (
              <span className="speech-recording-timer">
                {minutes}
:
                {seconds}
              </span>
            )
            : null}
          <div className="record-icon">
            <AudioRecorder
              recordIcon={StartRecordingIcon}
              stopIcon={StopRecordingIcon}
              iconProps={{
                stopProps: {
                  fill: '#004C97',
                },
                recordProps: {
                  fill: '#004C97',
                },
              }}
              styleOverride={{
                button: {
                  height: '70px',
                  ':focus': {
                    outline: 'none',
                  },
                  ':hover': {
                    boxShadow: 'none',
                  },
                },
              }}
              buttonHeight="70px"
              onRecord={this.onClickStartRecord}
              onStop={this.onClickStopRecord}
              registerCallbacks={this.registerCallbacks}
              dispatch={dispatch}
            />
          </div>
        </span>
      );
    case this.audioStates.beforePlaying:
    case this.audioStates.playing:
      return (
        <span>
          <div className="record-icon">
            <AudioPlayer
              audioSrc={audioSrc}
              playIcon={PlayRecordingIcon}
              stopIcon={StopRecordingIcon}
              iconProps={{
                stopProps: {
                  fill: '#004C97',
                },
                playProps: {
                  fill: '#004C97',
                },
              }}
              styleOverride={{
                button: {
                  height: '70px',
                  ':focus': {
                    outline: 'none',
                  },
                  ':hover': {
                    boxShadow: 'none',
                  },
                },
              }}
              onPlay={this.setAudioState}
              onStop={this.setAudioState}
              audioStates={this.audioStates}
              dispatch={dispatch}
            />
          </div>
        </span>
      );
    default:
      return '';
    }
  }

  render() {
    const { header } = this.props;
    const {
      isPlayButtonEnabled,
      isRunTestDisable,
    } = this.state;
    return (
      <div>
        <Dialog
          isOpen
          closeIconVisible
          onClickClose={this.onClickClose}
          okVisible
          okChildren={Constants.RUN_TEST}
          onClickOk={this.onClickRunTest}
          okDisabled={isRunTestDisable}
          cancelVisible={isPlayButtonEnabled}
          cancelChildren="RECORD AGAIN"
          onClickCancel={this.onClickRecordNew}
          headerChildren={header}
          centerContent
          styleOverride={this.styleOverride}
          showBodySeperator={false}
        >
          {this.renderChildren()}
          {this.getActionSubtitle()}
        </Dialog>
      </div>
    );
  }
}

RecordSpeechDialog.defaultProps = {
  header: '',
  onClickRunTest: () => {},
  dispatch: () => {},
};

RecordSpeechDialog.propTypes = {
  dispatch: PropTypes.func,
  header: PropTypes.node,
  onClickRunTest: PropTypes.func,
};
