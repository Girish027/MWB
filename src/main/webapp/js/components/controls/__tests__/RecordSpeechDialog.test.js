import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as actionsApp from 'state/actions/actions_app';
import Constants from 'constants/Constants';
import { getLanguage } from 'state/constants/getLanguage';
import RecordSpeechDialog from 'components/controls/RecordSpeechDialog';

jest.useFakeTimers();

describe('<RecordSpeechDialog />', () => {
  let wrapper;

  const props = {
    header: 'Record',
  };

  describe('Creating an instance with no props', () => {
    beforeEach(() => {
      wrapper = shallow(<RecordSpeechDialog />);
    });
    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    test('Snapshots should render correctly', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Creating an instance with test props', () => {
    beforeEach(() => {
      wrapper = shallow(<RecordSpeechDialog
        {...props}
      />);
      wrapper.setState({
        isRunTestDisable: true,
        isPlayButtonEnabled: false,
        seconds: 40,
        minutes: '00',
      });
      wrapper.update();
    });

    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    test('Snapshots should render correctly', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Various state of the Record Speech Dialog', () => {
    beforeEach(() => {
      wrapper = shallow(<RecordSpeechDialog
        {...props}
      />);
    });

    test('renders correctly when user has not recorded audio', () => {
      wrapper.setState({
        ...wrapper.instance().initialState,
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when user is recording audio', () => {
      wrapper.setState({
        audioState: wrapper.instance().audioStates.recording,
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when user has stopped recording audio', () => {
      wrapper.instance().getFileName = jest.fn(() => 'recording_1557955235157.wav');
      wrapper.instance().onClickStopRecord('audio data');
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when user has playing the recorded audio', () => {
      wrapper.instance().getFileName = jest.fn(() => 'recording_1557955235157.wav');

      wrapper.instance().onClickStopRecord('audio data');
      wrapper.instance().setAudioState(wrapper.instance().audioStates.playing);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when user has stopped playing the recorded audio', () => {
      wrapper.instance().getFileName = jest.fn(() => 'recording_1557955235157.wav');
      wrapper.instance().onClickStopRecord('audio data');
      wrapper.instance().setAudioState(wrapper.instance().audioStates.playing);
      wrapper.instance().setAudioState(wrapper.instance().audioStates.beforePlaying);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });


  describe('Functionality', () => {
    beforeAll(() => {
      props.dispatch = jest.fn();
      props.onClickRunTest = jest.fn();
      actionsApp.modalDialogChange = jest.fn(() => 'called modalDialogChange');
      actionsApp.displayWarningRequestMessage = jest.fn(() => 'called displayWarningRequestMessage');
    });

    beforeEach(() => {
      wrapper = shallow(<RecordSpeechDialog
        {...props}
      />);
    });

    describe('registerCallbacks', () => {
      test('should register the provided callback', () => {
        const callback = jest.fn();
        wrapper.instance().registerCallbacks({ callback });
        wrapper.instance().callbacks.callback();
        expect(callback).toHaveBeenCalled();
      });
    });

    describe('tick', () => {
      test('should decrement countdown seconds each time it is called', () => {
        wrapper.setState({
          seconds: 10,
        });
        wrapper.instance().tick();
        expect(wrapper.state().seconds).toEqual(9);
        wrapper.instance().tick();
        expect(wrapper.state().seconds).toEqual(8);
      });

      test('should clearInterval for the timer when countdown reaches zero', () => {
        wrapper.instance().timer = 20;
        wrapper.setState({
          seconds: 0,
        });
        wrapper.instance().tick();
        expect(clearInterval).toHaveBeenCalledWith(wrapper.instance().timer);
      });

      test('should call the stopMicrophone callback to stop currently active recording', () => {
        const stopMicrophone = jest.fn();
        wrapper.instance().registerCallbacks({ stopMicrophone });
        wrapper.setState({
          seconds: 0,
        });
        wrapper.instance().tick();
        expect(stopMicrophone).toHaveBeenCalled();
      });
    });

    test('should change the state when onClickStartRecord is clicked', () => {
      const wrapperInstance = wrapper.instance();
      wrapperInstance.onClickStartRecord();
      expect(wrapperInstance.state.audioState).toBe(wrapper.instance().audioStates.recording);
    });

    test('should start the timer when onClickStartRecord is clicked', () => {
      const wrapperInstance = wrapper.instance();
      wrapperInstance.onClickStartRecord();
      expect(setInterval).toHaveBeenLastCalledWith(wrapperInstance.tick, 1000);
    });

    test('should change the state when onClickStopRecord is clicked', () => {
      const wrapperInstance = wrapper.instance();
      wrapperInstance.onClickStopRecord();
      expect(wrapperInstance.state.isPlayButtonEnabled).toEqual(true);
      expect(wrapperInstance.state.isRunTestDisable).toEqual(false);
      expect(wrapperInstance.state.seconds).toEqual(Constants.RECORDING_MAX_TIME_LIMIT_SEC);
      expect(wrapperInstance.state.audioState).toBe(wrapper.instance().audioStates.beforePlaying);
    });

    test('should rest the state when onClickRecordNew is clicked', () => {
      const wrapperInstance = wrapper.instance();
      wrapperInstance.onClickRecordNew();
      expect(wrapperInstance.state).toEqual(wrapper.instance().initialState);
    });

    test('should decrement the counter', () => {
      const wrapperInstance = wrapper.instance();
      wrapperInstance.tick();
      expect(wrapperInstance.state.seconds).toBe(39);
    });

    test('should dispatch an action when onClickClose is clicked', () => {
      const wrapperInstance = wrapper.instance();
      wrapperInstance.onClickClose();
      expect(actionsApp.modalDialogChange).toHaveBeenCalledWith(null);
    });
    test('should reset state when onClickClose is clicked', () => {
      const wrapperInstance = wrapper.instance();
      wrapperInstance.onClickClose();
      expect(wrapperInstance.state).toEqual(wrapperInstance.initialState);
    });

    //    Need to debug this case and resolved.
    //    test('should show Progress Dialog when onClickRunTest is clicked', () => {
    //      const { dispatch } = props;
    //
    //      const wrapperInstance = wrapper.instance();
    //      wrapper.setState({
    //        audioFile: 'some File',
    //      });
    //      wrapperInstance.onClickRunTest();
    //
    //      expect(actionsApp.modalDialogChange).toHaveBeenCalledWith({
    //        dispatch,
    //        type: Constants.DIALOGS.PROGRESS_DIALOG,
    //        message: Constants.TEST_IN_PROGRESS,
    //      });
    //    });

    test('should call onClickRunTest of parent when onClickRunTest is clicked', () => {
      const { dispatch } = props;

      const wrapperInstance = wrapper.instance();
      wrapper.setState({
        audioFile: 'some audio file',
      });
      wrapperInstance.onClickRunTest();
      expect(props.onClickRunTest).toHaveBeenCalledWith(wrapper.state().audioFile);
    });

    test('should show warning if audio file is missing', () => {
      const { dispatch } = props;

      const wrapperInstance = wrapper.instance();
      wrapper.setState({
        audioFile: '',
      });
      wrapperInstance.onClickRunTest();
      const { DISPLAY_MESSAGES } = getLanguage();

      expect(actionsApp.displayWarningRequestMessage).toHaveBeenCalledWith(DISPLAY_MESSAGES.audioFileNotAvailable);
      expect(props.dispatch).toHaveBeenCalledWith('called displayWarningRequestMessage');
    });

    describe('getFileName:', () => {
      test('should get the name of the file with current timestamp', () => {
        expect(wrapper.instance().getFileName().match(/recording_\d+\.wav/)).not.toEqual(null);
      });
    });
  });
});
