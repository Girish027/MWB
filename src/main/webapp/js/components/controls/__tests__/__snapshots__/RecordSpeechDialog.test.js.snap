// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecordSpeechDialog /> Creating an instance with no props Snapshots should render correctly 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren="RECORD AGAIN"
    cancelVisible={false}
    centerContent={true}
    closeIconVisible={true}
    headerChildren=""
    isOpen={true}
    okChildren="RUN TEST"
    okDisabled={true}
    okVisible={true}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    showBodySeperator={false}
    styleOverride={
      Object {
        "cancel": Object {
          "backgroundColor": "unset",
          "paddingLeft": "10px",
          "paddingRight": "10px",
        },
        "childContainer": Object {
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "container": Object {
          "display": "grid",
          "gridTemplateRows": "60px auto 60px",
          "height": "340px",
          "width": "500px",
        },
        "ok": Object {
          "marginLeft": "10px",
          "paddingLeft": "25px",
          "paddingRight": "25px",
        },
      }
    }
  >
    <span>
      <div
        className="record-icon"
      >
        <AudioRecorder
          audioSrc=""
          buttonHeight="70px"
          dispatch={[Function]}
          iconProps={
            Object {
              "recordProps": Object {
                "fill": "#004C97",
              },
              "stopProps": Object {
                "fill": "#004C97",
              },
            }
          }
          onRecord={[Function]}
          onStop={[Function]}
          recordIcon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            }
          }
          registerCallbacks={[Function]}
          stopIcon={[Function]}
          styleOverride={
            Object {
              "button": Object {
                ":focus": Object {
                  "outline": "none",
                },
                ":hover": Object {
                  "boxShadow": "none",
                },
                "height": "70px",
              },
            }
          }
        />
      </div>
    </span>
    <span
      className="speech-recording-span blueText"
    >
      START
    </span>
  </withTheme(DialogComponent)>
</div>
`;

exports[`<RecordSpeechDialog /> Creating an instance with test props Snapshots should render correctly 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren="RECORD AGAIN"
    cancelVisible={false}
    centerContent={true}
    closeIconVisible={true}
    headerChildren="Record"
    isOpen={true}
    okChildren="RUN TEST"
    okDisabled={true}
    okVisible={true}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    showBodySeperator={false}
    styleOverride={
      Object {
        "cancel": Object {
          "backgroundColor": "unset",
          "paddingLeft": "10px",
          "paddingRight": "10px",
        },
        "childContainer": Object {
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "container": Object {
          "display": "grid",
          "gridTemplateRows": "60px auto 60px",
          "height": "340px",
          "width": "500px",
        },
        "ok": Object {
          "marginLeft": "10px",
          "paddingLeft": "25px",
          "paddingRight": "25px",
        },
      }
    }
  >
    <span>
      <div
        className="record-icon"
      >
        <AudioRecorder
          audioSrc=""
          buttonHeight="70px"
          dispatch={[Function]}
          iconProps={
            Object {
              "recordProps": Object {
                "fill": "#004C97",
              },
              "stopProps": Object {
                "fill": "#004C97",
              },
            }
          }
          onRecord={[Function]}
          onStop={[Function]}
          recordIcon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            }
          }
          registerCallbacks={[Function]}
          stopIcon={[Function]}
          styleOverride={
            Object {
              "button": Object {
                ":focus": Object {
                  "outline": "none",
                },
                ":hover": Object {
                  "boxShadow": "none",
                },
                "height": "70px",
              },
            }
          }
        />
      </div>
    </span>
    <span
      className="speech-recording-span blueText"
    >
      START
    </span>
  </withTheme(DialogComponent)>
</div>
`;

exports[`<RecordSpeechDialog /> Various state of the Record Speech Dialog renders correctly when user has not recorded audio 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren="RECORD AGAIN"
    cancelVisible={false}
    centerContent={true}
    closeIconVisible={true}
    headerChildren="Record"
    isOpen={true}
    okChildren="RUN TEST"
    okDisabled={true}
    okVisible={true}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    showBodySeperator={false}
    styleOverride={
      Object {
        "cancel": Object {
          "backgroundColor": "unset",
          "paddingLeft": "10px",
          "paddingRight": "10px",
        },
        "childContainer": Object {
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "container": Object {
          "display": "grid",
          "gridTemplateRows": "60px auto 60px",
          "height": "340px",
          "width": "500px",
        },
        "ok": Object {
          "marginLeft": "10px",
          "paddingLeft": "25px",
          "paddingRight": "25px",
        },
      }
    }
  >
    <span>
      <div
        className="record-icon"
      >
        <AudioRecorder
          audioSrc=""
          buttonHeight="70px"
          dispatch={[Function]}
          iconProps={
            Object {
              "recordProps": Object {
                "fill": "#004C97",
              },
              "stopProps": Object {
                "fill": "#004C97",
              },
            }
          }
          onRecord={[Function]}
          onStop={[Function]}
          recordIcon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            }
          }
          registerCallbacks={[Function]}
          stopIcon={[Function]}
          styleOverride={
            Object {
              "button": Object {
                ":focus": Object {
                  "outline": "none",
                },
                ":hover": Object {
                  "boxShadow": "none",
                },
                "height": "70px",
              },
            }
          }
        />
      </div>
    </span>
    <span
      className="speech-recording-span blueText"
    >
      START
    </span>
  </withTheme(DialogComponent)>
</div>
`;

exports[`<RecordSpeechDialog /> Various state of the Record Speech Dialog renders correctly when user has playing the recorded audio 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren="RECORD AGAIN"
    cancelVisible={true}
    centerContent={true}
    closeIconVisible={true}
    headerChildren="Record"
    isOpen={true}
    okChildren="RUN TEST"
    okDisabled={false}
    okVisible={true}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    showBodySeperator={false}
    styleOverride={
      Object {
        "cancel": Object {
          "backgroundColor": "unset",
          "paddingLeft": "10px",
          "paddingRight": "10px",
        },
        "childContainer": Object {
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "container": Object {
          "display": "grid",
          "gridTemplateRows": "60px auto 60px",
          "height": "340px",
          "width": "500px",
        },
        "ok": Object {
          "marginLeft": "10px",
          "paddingLeft": "25px",
          "paddingRight": "25px",
        },
      }
    }
  >
    <span>
      <div
        className="record-icon"
      >
        <AudioPlayer
          audioSrc="file"
          audioStates={
            Object {
              "beforePlaying": "beforePlaying",
              "beforeRecording": "beforeRecording",
              "playing": "playing",
              "recording": "recording",
            }
          }
          dispatch={[Function]}
          iconProps={
            Object {
              "playProps": Object {
                "fill": "#004C97",
              },
              "stopProps": Object {
                "fill": "#004C97",
              },
            }
          }
          onPlay={[Function]}
          onStop={[Function]}
          playIcon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            }
          }
          stopIcon={[Function]}
          styleOverride={
            Object {
              "button": Object {
                ":focus": Object {
                  "outline": "none",
                },
                ":hover": Object {
                  "boxShadow": "none",
                },
                "height": "70px",
              },
            }
          }
        />
      </div>
    </span>
    <span
      className="speech-recording-span blueText"
    >
      STOP
      <span
        style={
          Object {
            "paddingLeft": "10px",
          }
        }
      >
        <PrimaryNavDivider
          fill="#004C97"
          float="left"
          height="20px"
        />
        <Downloader
          dispatch={[Function]}
          file="file"
          fileName="recording_1557955235157.wav"
          fileType="recording"
          iconProps={Object {}}
        >
          DOWNLOAD
        </Downloader>
      </span>
    </span>
  </withTheme(DialogComponent)>
</div>
`;

exports[`<RecordSpeechDialog /> Various state of the Record Speech Dialog renders correctly when user has stopped playing the recorded audio 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren="RECORD AGAIN"
    cancelVisible={true}
    centerContent={true}
    closeIconVisible={true}
    headerChildren="Record"
    isOpen={true}
    okChildren="RUN TEST"
    okDisabled={false}
    okVisible={true}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    showBodySeperator={false}
    styleOverride={
      Object {
        "cancel": Object {
          "backgroundColor": "unset",
          "paddingLeft": "10px",
          "paddingRight": "10px",
        },
        "childContainer": Object {
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "container": Object {
          "display": "grid",
          "gridTemplateRows": "60px auto 60px",
          "height": "340px",
          "width": "500px",
        },
        "ok": Object {
          "marginLeft": "10px",
          "paddingLeft": "25px",
          "paddingRight": "25px",
        },
      }
    }
  >
    <span>
      <div
        className="record-icon"
      >
        <AudioPlayer
          audioSrc="file"
          audioStates={
            Object {
              "beforePlaying": "beforePlaying",
              "beforeRecording": "beforeRecording",
              "playing": "playing",
              "recording": "recording",
            }
          }
          dispatch={[Function]}
          iconProps={
            Object {
              "playProps": Object {
                "fill": "#004C97",
              },
              "stopProps": Object {
                "fill": "#004C97",
              },
            }
          }
          onPlay={[Function]}
          onStop={[Function]}
          playIcon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            }
          }
          stopIcon={[Function]}
          styleOverride={
            Object {
              "button": Object {
                ":focus": Object {
                  "outline": "none",
                },
                ":hover": Object {
                  "boxShadow": "none",
                },
                "height": "70px",
              },
            }
          }
        />
      </div>
    </span>
    <span
      className="speech-recording-span blueText"
    >
      PLAY
      <span
        style={
          Object {
            "paddingLeft": "10px",
          }
        }
      >
        <PrimaryNavDivider
          fill="#004C97"
          float="left"
          height="20px"
        />
        <Downloader
          dispatch={[Function]}
          file="file"
          fileName="recording_1557955235157.wav"
          fileType="recording"
          iconProps={Object {}}
        >
          DOWNLOAD
        </Downloader>
      </span>
    </span>
  </withTheme(DialogComponent)>
</div>
`;

exports[`<RecordSpeechDialog /> Various state of the Record Speech Dialog renders correctly when user has stopped recording audio 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren="RECORD AGAIN"
    cancelVisible={true}
    centerContent={true}
    closeIconVisible={true}
    headerChildren="Record"
    isOpen={true}
    okChildren="RUN TEST"
    okDisabled={false}
    okVisible={true}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    showBodySeperator={false}
    styleOverride={
      Object {
        "cancel": Object {
          "backgroundColor": "unset",
          "paddingLeft": "10px",
          "paddingRight": "10px",
        },
        "childContainer": Object {
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "container": Object {
          "display": "grid",
          "gridTemplateRows": "60px auto 60px",
          "height": "340px",
          "width": "500px",
        },
        "ok": Object {
          "marginLeft": "10px",
          "paddingLeft": "25px",
          "paddingRight": "25px",
        },
      }
    }
  >
    <span>
      <div
        className="record-icon"
      >
        <AudioPlayer
          audioSrc="file"
          audioStates={
            Object {
              "beforePlaying": "beforePlaying",
              "beforeRecording": "beforeRecording",
              "playing": "playing",
              "recording": "recording",
            }
          }
          dispatch={[Function]}
          iconProps={
            Object {
              "playProps": Object {
                "fill": "#004C97",
              },
              "stopProps": Object {
                "fill": "#004C97",
              },
            }
          }
          onPlay={[Function]}
          onStop={[Function]}
          playIcon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            }
          }
          stopIcon={[Function]}
          styleOverride={
            Object {
              "button": Object {
                ":focus": Object {
                  "outline": "none",
                },
                ":hover": Object {
                  "boxShadow": "none",
                },
                "height": "70px",
              },
            }
          }
        />
      </div>
    </span>
    <span
      className="speech-recording-span blueText"
    >
      PLAY
      <span
        style={
          Object {
            "paddingLeft": "10px",
          }
        }
      >
        <PrimaryNavDivider
          fill="#004C97"
          float="left"
          height="20px"
        />
        <Downloader
          dispatch={[Function]}
          file="file"
          fileName="recording_1557955235157.wav"
          fileType="recording"
          iconProps={Object {}}
        >
          DOWNLOAD
        </Downloader>
      </span>
    </span>
  </withTheme(DialogComponent)>
</div>
`;

exports[`<RecordSpeechDialog /> Various state of the Record Speech Dialog renders correctly when user is recording audio 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren="RECORD AGAIN"
    cancelVisible={false}
    centerContent={true}
    closeIconVisible={true}
    headerChildren="Record"
    isOpen={true}
    okChildren="RUN TEST"
    okDisabled={true}
    okVisible={true}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    showBodySeperator={false}
    styleOverride={
      Object {
        "cancel": Object {
          "backgroundColor": "unset",
          "paddingLeft": "10px",
          "paddingRight": "10px",
        },
        "childContainer": Object {
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "container": Object {
          "display": "grid",
          "gridTemplateRows": "60px auto 60px",
          "height": "340px",
          "width": "500px",
        },
        "ok": Object {
          "marginLeft": "10px",
          "paddingLeft": "25px",
          "paddingRight": "25px",
        },
      }
    }
  >
    <span>
      <span
        className="speech-recording-timer"
      >
        00
        :
        40
      </span>
      <div
        className="record-icon"
      >
        <AudioRecorder
          audioSrc=""
          buttonHeight="70px"
          dispatch={[Function]}
          iconProps={
            Object {
              "recordProps": Object {
                "fill": "#004C97",
              },
              "stopProps": Object {
                "fill": "#004C97",
              },
            }
          }
          onRecord={[Function]}
          onStop={[Function]}
          recordIcon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            }
          }
          registerCallbacks={[Function]}
          stopIcon={[Function]}
          styleOverride={
            Object {
              "button": Object {
                ":focus": Object {
                  "outline": "none",
                },
                ":hover": Object {
                  "boxShadow": "none",
                },
                "height": "70px",
              },
            }
          }
        />
      </div>
    </span>
    <span
      className="speech-recording-span blueText"
    >
      STOP
    </span>
  </withTheme(DialogComponent)>
</div>
`;
